frameworks/supervised/exec.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+import tempfile as tmp
+import pandas as pd
+import numpy as np
+
+from frameworks.shared.callee import call_run, result, Timer, touch
+
+log = logging.getLogger(os.path.basename(__file__))
+
+import supervised
+from supervised.automl import AutoML
+import shutil
+
+
+def make_subdir(name, config):
+    subdir = os.path.join(config.output_dir, name, config.name, str(config.fold))
+    touch(subdir, as_dir=True)
+    return subdir
+
+def run(dataset, config):
+    log.info("\n**** mljar-supervised ****\n")
+
+    is_classification = config.type == "classification"
+
+    X_train, X_test = dataset.train.X, dataset.test.X
+    y_train, y_test = dataset.train.y, dataset.test.y
+
+    X_train = pd.DataFrame(X_train, columns=[f"f{i}" for i in range(X_train.shape[1])])
+    X_test = pd.DataFrame(X_test, columns=[f"f{i}" for i in range(X_test.shape[1])])
+
+    y_train = y_train.flatten()
+    y_test = y_test.flatten()
+
+
+    # cast columns to have float types
+    for col in X_train.columns:
+        try: 
+            x1 = X_train[col].astype(float)
+            X_train[col] = x1
+            x2 = X_test[col].astype(float)
+            X_test[col] = x2
+        except Exception as e:
```

Comment:
does it mean you're ignoring all categorical columns?

Reply:
if so, why no using encoded data instead, like for `autosklearn` for example.

Reply:
Thank you for review! :) 

The problem is that all columns have `object` type - all looks like categoricals. That's why I'm trying to cast into `float` type. If they can't be cast then they are left as `object`. My framework distinguish categoricals and numeric, it is using pandas types for this.

Reply:
gotcha!
if you want a bit cleaner way to do this, you can previously list all columns that are categoricals, see https://github.com/openml/automlbenchmark/blob/master/frameworks/autosklearn/__init__.py

Reply:
from what I remember all columns are `object` type that's why loop over all of them

Reply:
`object` type is at `numpy` level, but when parsing the data file, especially if it's in ARFF format, we can retrieve the "real" type, this is why the snippet above can distinguish categoricals from numericals.

Reply:
typically this would be more explicit on your side:
in `__init__`:
```
   data = dict(
        train=dict(
            X=dataset.train.X,
            y=dataset.train.y
        ),
        ...
        column_names = [p.name for p in dataset.predictors]
        column_type={p.name: ('object' if p.is_categorical() else 'float') for p in dataset.predictors}
    )
```
in `exec`:
```
X_train = pd.DataFrame(X_train, columns=data.column_names).astype(column_types)
```

Reply:
@sebhrusen thank you! I've used your example, code looks much better.